File: src/repositories/session-count.js
// Append-only log (session_count) — one row per session a user engages
// with, fired on the first card answered for that session_number/day. Same
// conventions as the other choices-app log tables: user_id is the real Eaze
// platform user id, only recorded when the caller actually has one
// (banner-entered users), never fabricated for the typed-phone dev login
// path.
import { dbEnabled, query } from '../db/client.js';
import { istNowParts } from '../lib/ist-time.js';

export async function recordSessionEngagement({ userId, phoneNumber, sessionNumber }) {
  if (!dbEnabled) return { persisted: false };
  const { date, time } = istNowParts();
  const { rows } = await query(
    `INSERT INTO session_count (user_id, phone_number, session_number, log_date, log_time)
     VALUES ($1, $2, $3, $4, $5)
     RETURNING id, user_id, phone_number, session_number, log_date, log_time`,
    [userId, phoneNumber, sessionNumber, date, time],
  );
  return { persisted: true, log: rows[0] };
}

File: app/repositories/session_count.py
# Append-only log (session_count) — one row per session a user engages
# with, fired on the first card answered for that session_number/day. Same
# conventions as the other choices-app log tables: user_id is the real Eaze
# platform user id, only recorded when the caller actually has one
# (banner-entered users), never fabricated for the typed-phone dev login
# path.
from ..db.client import DB_ENABLED, query
from ..lib.ist_time import ist_now_parts


async def record_session_engagement(*, user_id, phone_number, session_number):
    if not DB_ENABLED:
        return {"persisted": False}
    parts = ist_now_parts()
    result = await query(
        """INSERT INTO session_count (user_id, phone_number, session_number, log_date, log_time)
           VALUES (%s, %s, %s, %s, %s)
           RETURNING id, user_id, phone_number, session_number, log_date, log_time""",
        [user_id, phone_number, session_number, parts["date"], parts["time"]],
    )
    return {"persisted": True, "log": result["rows"][0]}
